src/group_theory/group_action/basic.lean

```diff
@@ -530,6 +530,10 @@ lemma finset.smul_sum {r : α} {f : γ → β} {s : finset γ} :
   r • ∑ x in s, f x = ∑ x in s, r • f x :=
 (distrib_mul_action.to_add_monoid_hom β r).map_sum f s
 
+lemma finset.sum_smul_sum {α : Type*} [semiring α] [module α β] {f : γ → α} {g : γ → β}
```

Comment:
This file feels too early to be referencing `module`; I'm surprised it's even defined yet here.

Reply:
Well, I wanted to put it in `algebra.module.basic` but that is cross-imported by `group_theory.group_action.basic`, so I couldn't really do that

Reply:
Currently, `group.theory.subgroup.basic` imports `algebra.module.basic` yet removing this import causes no problems in this file, so I think that is the way to go. I would then put my lemma in `algebra.module.basic`. @eric-wieser

Reply:
That sounds good to me, thanks!

Reply:
Ah, I wish it were that easy. It looks like `group.theory.submonoid.operations` imports `data.set.pointwise`, which imports `algebra.module.basic` (I think it really has to for both imports), yet if we choose to import `group_theory.group_action.basic` to `algebra.module_basic`, we obviously have a cycle. Do you have an idea of how to fix this? I might be wrong but I think it'd be nice to have all of `finset.smul_sum`, `finset.sum_smul` and `finset.sum_smul_sum` in the same file @eric-wieser

Reply:
I created #13340 to try and ease things here. I think the underlying problem is that `group_action.basic` is not trivial results about group actions, it's a bunch of orbit stabilizer stuff that pulls in various bits of mathlib.

I don't think we want all of those in the same file, as the last two have stronger assumptions than the first one.